core/src/main/scala/cats/Foldable.scala

```diff
@@ -587,11 +594,14 @@ import Foldable.sentinel
 object Foldable {
   private val sentinel: Function1[Any, Any] = new scala.runtime.AbstractFunction1[Any, Any] { def apply(a: Any) = this }
 
-  def iterateRight[A, B](iterable: Iterable[A], lb: Eval[B])(f: (A, Eval[B]) => Eval[B]): Eval[B] = {
-    def loop(it: Iterator[A]): Eval[B] =
-      Eval.defer(if (it.hasNext) f(it.next, loop(it)) else lb)
+  def iterateRight[A, B](iterable: Iterable[A], lb: Eval[B])(f: (A, Eval[B]) => Eval[B]): Eval[B] =
+    iterateRightDefer(iterable, lb)(f)
```

Comment:
I would imagine inlining `Eval.defer` will give a slight performance improvement since the jit can at best do this inlining later.

Reply:
@johnynek not sure I understand. Are you saying you would prefer the previous implementation explicitly calling `Eval.defer`?